src/utils.c

```diff
@@ -2152,39 +2157,84 @@ const gchar *utils_resource_dir(GeanyResourceDirType type)
 }
 
 
+/* "Reverse parse" GOptionEntry */
+static gchar *utils_option_entry_reverse_parse(const GOptionEntry *optentry)
+{
+	gchar *s = NULL;
+
+	switch (optentry->arg)
+	{
+		case G_OPTION_ARG_NONE:
+			{
+				gboolean val = *(gboolean *)optentry->arg_data;
+				gboolean reverse = (optentry->flags & G_OPTION_FLAG_REVERSE);
+				if ((val && !reverse) || (!val && reverse)) /* logical XOR */
+					s = g_strdup_printf("--%s", optentry->long_name);
+			}
+			break;
+
+		case G_OPTION_ARG_INT:
+			if (*(gint *)optentry->arg_data)
+				s = g_strdup_printf("--%s=%d", optentry->long_name, *(gint *)optentry->arg_data);
+			break;
+
+		case G_OPTION_ARG_STRING:
+		case G_OPTION_ARG_FILENAME:
+			if (*(gchar **)optentry->arg_data)
+				s = g_strdup_printf("--%s=%s", optentry->long_name, *(gchar **)optentry->arg_data);
+			break;
+
+		default:
+			g_warning("%s: %s: %d\n", G_STRFUNC, "Unsupported option entry type", optentry->arg);
+	}
+
+	return s;
+}
+
+
 void utils_start_new_geany_instance(const gchar *doc_path)
 {
-	const gchar *command = is_osx_bundle() ? "open" : "geany";
-	gchar *exec_path = g_find_program_in_path(command);
+	gchar **argv;
+	int i, persistent;
+	int argc;
+	GError *error = NULL;
+
+	for (i = 0, persistent = 0; optentries[i].long_name; i++)
+		if (optentries_aux[i].persistent)
+			persistent++;
+
+	argv = g_new0(gchar *, persistent + 5);  /* exectuable, -i, --, doc_path, NULL */
+	argv[0] = g_build_filename(utils_resource_dir(RESOURCE_DIR_PREFIX), "bin", "geany", NULL);
```

Comment:
It is. Since I don't want an unneeded RESOURCE_DIR_PREFIX under OSX, and g_build_filename() scattered in utils_resource_dir() and utils_start_new_geany_instance(), I'll do the following:

- Rename utils_resource_dir() to utils_resource_path().
- Rename RESOURCE_DIR_PREFIX to RESOURCE_FILE_GEANY or something (please suggest a name).
- Alter utils_resource_path(RESOURCE_FILE_GEANY) to return gtkosx_application_get_executable_path() under OSX or prefix/bin/geany otherwise.

It is the executable name that we really need from utils_resource_path(), not the RESOURCE_DIR_PREFIX.